code/modules/projectiles/projectile/energy/thermal.dm

```diff
@@ -10,10 +10,10 @@
 	bare_wound_bonus = 10
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser
 
-/obj/projectile/energy/inferno/on_hit(atom/target, blocked, pierce_hit)
-	..()
+/obj/projectile/energy/inferno/on_hit(atom/target, blocked = 0, pierce_hit)
+	. = ..()
 	if(!ishuman(target))
-		return
+		return ..()
```

Comment:
Now hold on I think you want to change _either_ of these lines but not _both_
Now you're calling the parent proc twice in both of these instances

Reply:
Ah I think you're right sorry it's late for me